### ISSUE TITLE
Weird z-sorting with multiple `<Html occlusion={'blending'} />`

### ISSUE THREAD

**Issue by FarazzShaikh** [pmndrs/drei#1255]
- `@react-three/drei` version: 9.56.5

### Problem description:

When using multiple overlapping `<Html />` with blending occlusion, they are not z-sorted accurately. 

https://user-images.githubusercontent.com/55190601/215278498-f16a742c-d73b-42f5-986c-2e64e371dd43.mp4

### Relevant code:

https://codesandbox.io/s/mixing-html-and-webgl-w-occlusion-forked-7qbp7l?file=/src/App.js

### Suggested solution:

Donno, we cannot perform z-sorting per-pixel on the HTML elements, thus, there will always need to be a threshold where the "bottom" HTML pops over the "top", bellow this threshold, overlapping HTML will look broken.

**Comment by sclea** [comment]
@FarazzShaikh It seems the problem is somewhere else. I has experimented just now and find the interesting thing.
Adding rotation-y={1} to «HTML /» [makes it work](https://codesandbox.io/s/mixing-html-and-webgl-w-occlusion-forked-7qbp7l?file=/src/App.js) like a charm, just like it have to be (see htmls with blue and red bgs).
Maybe there's some delta somewhere added to Y axis that leads to wrong z-index calculation?
I beilive I rougly understand the math behind but I don't even know where it's located in code ^)

The second thing that we can't fix I'm afraid -- that is crossing htmls (green bg). Have no any idea except to render html to canvas and use masking but we lose interactivity this way.
Well maybe it's no real life use case for this at all?

**Comment by FarazzShaikh** [comment]
The issue still prersists in your box, its jsut less notacable 

https://user-images.githubusercontent.com/55190601/215324111-27966ccc-0daf-45ef-ac90-e1ad07f9cdc0.mp4

Here is the place in the code where the zIndex is being set: https://github.com/pmndrs/drei/blob/989452359832af94f78363385412572166d028af/src/web/Html.tsx#L70-L80

Here is where this function is used:  https://github.com/pmndrs/drei/blob/989452359832af94f78363385412572166d028af/src/web/Html.tsx#L318

**Comment by sclea** [comment]
> The issue still prersists in your box, its jsut less notacable

Yes, I saw it, but it is for 90 degrees rotated htmls (green + (red | blue)), it's okay and it is unfixable with z-index of course.

I meant that red and blue now (with rotation-y={1}) renders correct.

**Comment by dberardo-com** [comment]
any updates on this ... i also get a strange behavior with occlude blending ... also seems like pointer events are sort of not working correctly with blending occlusion

**Comment by m4cn4ir** [comment]
Hi,

I would also like to know if there is any progress on this as I am experiencing issues in my live gallery. I have two problems with html. Its also related to this thread: https://github.com/pmndrs/drei/issues/1228

1. z-index issues:
In my gallery, images (textures as three meshes) and youtube videos (html with occlude=blending) can be shown. Initially, these coexist perfectly and the youtube videos hide behind the walls and other images. When another player loggs in, his avatar and his nametag will be rendered. The nametag is also html but without occlusion. And this mixing of html with different types of occlusion will lead to z-index issues. All the youtube videos will now shine through walls and other meshes. I can "fix" this by rerendering the videos component manually. I fiddled with the zIndexRange but can't seem to fix it with this approach. Of course when I rerender the videos component, the video playback also stops which is quite bad for UX. Is there a way to manually call this "z-index recalculation" without rerendering the entire component until the issue is resolved?

2. raycasting issues
My images and youtube video frames have onPointerEnter hover handles and show <Outline ../> when hovering. Raycasting and hovering of the 3d meshes works flawless but as soon as I add html youtube videos to my scene, weird things start to happen. First of all, hovering over the html videos frame, sometimes the events pass through. It is as if there were holes in the frame. Also, sometimes when hovering the html video, the raycaster recognizes other 3d meshes as being hovered! Of course this sounds like a logic flaw but it really is not.

I didnt provide a sandbox because I think that these issues are related and the current material should be enough. But I would be happy to provide more if this helps for this issue. Thanks!

**Comment by m4cn4ir** [comment]
I would like to bump this and prevent it from being closed. Mixing HTML with different occlusions should be a usecase in multiple applications.
Bounding boxes and raycasting seem to work initially, break after something happens and then can be fixed again by forcing rerenders. There should be an easy solution to this or at least a quick fix to manually trigger this.

**Comment by 0HammerHead0** [comment]
I am experiencing similar issues where any mesh ( doesn't matter if it is behind or in front of the Html ) cuts the intersected section of the Html.
I don't have any flickering issues.
For blending, it cuts the intersected area and for raycast, it is completely invisible.


PS: i am using it with imported GLB models, nothing special.
